common/web/keyboard-processor/src/text/kbdInterface.ts

```diff
@@ -529,6 +529,43 @@ namespace com.keyman.text {
       }
     }
 
+    /**
+     * Maps a KeyEvent's modifiers to their appropriate value for key-rule evaluation
+     * based on the rule's specified target modifier set.
+     *
+     * Mostly used to correct chiral OSK-keys targeting non-chiral rules.
+     * @param e The source KeyEvent
+     * @returns
+     */
+    private static getEventRuleModifiers(e: KeyEvent, targetModifierMask: number): number {
+      const CHIRAL_ALT  = Codes.modifierCodes["LALT"]  | Codes.modifierCodes["RALT"];
+      const CHIRAL_CTRL = Codes.modifierCodes["LCTRL"] | Codes.modifierCodes["RCTRL"];
+
+      let modifiers = e.Lmodifiers;
```

Comment:
```suggestion
      let modifiers = eventModifiers;
```

If you go with my earlier suggestion... (Or, remove the `modifiers` variable altogether and just work with the `eventModifiers` input parameter, which is probably better.)